src/main/telemetry/smartport.c

```diff
@@ -85,15 +110,46 @@ enum
     FSSP_DATAID_HOME_DIST  = 0x0420 ,
     FSSP_DATAID_GPS_ALT    = 0x0820 ,
     FSSP_DATAID_ASPD       = 0x0A00 ,
+    FSSP_DATAID_TEMP       = 0x0B70 ,
+    FSSP_DATAID_TEMP1      = 0x0B71 ,
+    FSSP_DATAID_TEMP2      = 0x0B72 ,
+    FSSP_DATAID_TEMP3      = 0x0B73 ,
+    FSSP_DATAID_TEMP4      = 0x0B74 ,
+    FSSP_DATAID_TEMP5      = 0x0B75 ,
+    FSSP_DATAID_TEMP6      = 0x0B76 ,
+    FSSP_DATAID_TEMP7      = 0x0B77 ,
+    FSSP_DATAID_TEMP8      = 0x0B78 ,
     FSSP_DATAID_A3         = 0x0900 ,
     FSSP_DATAID_A4         = 0x0910
 };
 
+/*
+ * Row can be:
+ *  - single data ID: dataId
+ *  - range of data IDs: startDataId, count
+ *    the range is from startDataId to startDataId + count
+ *    where count can be > 1 and <= 16
+ *    or 1 for getMotorCount()
+ * A dataId (or count) of 0 indicates the end of the table.
+ */
+#define DATAID_TABLE_END     0
+#define MOTORS_DATAID_COUNT  1
 const uint16_t frSkyDataIdTable[] = {
     FSSP_DATAID_SPEED     ,
     FSSP_DATAID_VFAS      ,
+#ifdef USE_ESC_SENSOR
+    FSSP_DATAID_VFAS1     , MOTORS_DATAID_COUNT ,
```

Comment:
I don't like how this table is used to encode a mix of elements represented by single entries, and elements represented by a pair of entries. This is brittle and hard to understand. What if somebody wanted to add a sensor in the range [1, 16]? Also, it uses `uint16_t` to encode a value < 16. We don't have a case using an arbitrary number of sensors cycled in one slot in the queue, just `getMotorCount()`, so adding this capability is purely speculative.
How about we use a second array for the sensor groups that need to be cycled by motor id, and loop through this after looping through `frSkyDataIdTable`?

Reply:
From what I can gather, data IDs always have a non-zero high order byte so there should be no chance of collision between them and the count. Yes, two arrays would be much simpler, I was unsure whether the current order of data IDs was important or not.